source/tyr/tyr/resources.py

```diff
@@ -704,34 +711,52 @@ def get(self, user_id=None):
             g.disable_geojson = args['disable_geojson']
             user = models.User.query.get_or_404(user_id)
 
-            return marshal(user, user_fields_full)
+            return marshal(user, user_fields_full), None
         else:
             parser.add_argument('login', type=unicode, required=False, case_sensitive=False, help='login')
             parser.add_argument('email', type=unicode, required=False, case_sensitive=False, help='email')
             parser.add_argument('key', type=unicode, required=False, case_sensitive=False, help='key')
             parser.add_argument('end_point_id', type=int)
             parser.add_argument('block_until', type=datetime_format, required=False, case_sensitive=False)
+            parser.add_argument('page', type=int, required=False, default=1)
 
             args = parser.parse_args()
             g.disable_geojson = args['disable_geojson']
 
             if args['key']:
                 logging.debug(args['key'])
                 users = models.User.get_from_token(args['key'], datetime.now())
-                return marshal(users, user_fields)
+                return marshal(users, user_fields), None
             else:
                 del args['disable_geojson']
                 # dict comprehension would be better, but it's not in python 2.6
-                filter_params = dict((k, v) for k, v in args.items() if v)
+                filter_params = dict((k, v) for k, v in args.items() if v and k != 'page')
 
-                if filter_params:
-                    users = models.User.query.filter_by(**filter_params).all()
-                    return marshal(users, user_fields)
-                else:
-                    users = models.User.query.all()
-                    return marshal(users, user_fields)
-
-    def post(self):
+            if version == 1:
```

Comment:
This increases cyclomatic complexity to deal with all in the same file.
Maybe it's too costy, but I would like more a "dry" version of abstract-resources, inherited (and reformated) by v0-resources and v1-resources.
That way if we ever want to unplug v0 it should be easier just remove v0-resources file)

> a bit like what was done for COTS and IRE, even if Kirin is not a model on cyclomatic complexity :wink:

Reply:
To be discussed: as it 's the second time this comment comes up, I'll be fine with it